crates/db_views/src/comment_view.rs

```diff
@@ -190,10 +190,20 @@ fn queries<'a>() -> Queries<
       query = query.filter(comment_like::score.eq(-1));
     }
 
-    let is_creator = options.creator_id == options.local_user.map(|l| l.person.id);
-    // only show deleted comments to creator
-    if !is_creator {
-      query = query.filter(comment::deleted.eq(false));
+    // only show deleted comments to creator, or if they have children
+    if let Some(local_user) = options.local_user {
+      query = query.filter(
+        comment::deleted
+          .eq(false)
+          .or(comment::creator_id.eq(local_user.person.id))
+          .or(comment_aggregates::child_count.gt(0)),
+      );
+    } else {
+      query = query.filter(
+        comment::deleted
+          .eq(false)
+          .or(comment_aggregates::child_count.gt(0)),
+      );
```

Comment:
The old logic for this only showed deleted comments to the creator of the comment when creator ID was specified. This mean the only way to see them is if you use the search page and filter by creator, or if you are viewing your own profile. It also had the likely unintentional side effect of showing deleted comments to unauthenticated users. For example:
![comment-fix-before-unauthenticated](https://github.com/LemmyNet/lemmy/assets/28871516/99bbea33-9350-4576-8344-7c2dcf8ba1b8)

Reply:
Can you add a test case to make sure this wont get broken in the future again? Either a Rust test in the same file, or a Typescript test under `api_tests` folder.

Reply:
There are 2 blocks that check for *deleted* and *removed* in this file. IMO those blocks should be entirely removed... its too difficult and pointless to try to remove data in trees, and still have the rest of the tree render. I'm not sure why I would have approved adding those for comment_views. 

It should be left to front ends to make sure to render deleted and removed (IE show that they're deleted / removed, and don't show the content).

Reply:
It looks like the logic was originally added around a year ago.

![image](https://github.com/LemmyNet/lemmy/assets/28871516/2bbad2a2-466b-4e9e-ac02-5a8181ed37fd)